internal/grpc/services/storageprovider/storageprovider.go

```diff
@@ -867,6 +891,12 @@ func (s *service) ListRecycleStream(req *provider.ListRecycleStreamRequest, ss p
 
 	// TODO(labkode): CRITICAL: fill recycle info with storage provider.
 	for _, item := range items {
+		if item.Ref != nil && item.Ref.ResourceId != nil {
+			// The storage driver might set the mount ID by itself, in which case skip this step
+			if mountID, _ := storagespace.SplitStorageID(item.Ref.GetResourceId().GetStorageId()); mountID == "" {
+				item.Ref.ResourceId.StorageId = storagespace.FormatStorageID(s.conf.MountID, item.Ref.GetResourceId().GetStorageId())
+			}
+		}
```

Comment:
maybe extract this as `addMissingStorageProviderID` and reuse it everywhere in this file?